tests/NuGet.Services.AzureSearch.FunctionalTests/BasicTests/V2SearchProtocolTests.cs

```diff
@@ -0,0 +1,266 @@
+                        ? DateTime.Compare((DateTime)x1, (DateTime)x2) >= 0
+                        : string.Compare(x1.ToString(), x2.ToString()) >= 0;
+                };
+
+            for (int i = 1; i < topResults.Count(); i++)
+            {
+                if (reverse)
+                {
+                    // flip for ascending comparison
+                    Assert.True(comparer(topResults[i], topResults[i-1]));
+                } else
+                {
+                    Assert.True(comparer(topResults[i - 1], topResults[i]));
+                }
+            }
+        }
+
+        [Fact]
+        public async Task ResultsMatchIdQueryFieldSearch()
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "packageId:" + Constants.TestPackageId });
+
+            Assert.NotNull(results);
+            Assert.True(results.Data.Count == 1);
+            Assert.Equal(Constants.TestPackageId, results.Data.ToList()[0].PackageRegistration.Id, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public async Task ResultsMatchAuthorQueryFieldSearch()
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "author:" + Constants.TestPackageAuthor });
+
+            Assert.NotNull(results);
+
+            var authorResults = results
+                .Data
+                .Select(x => x.Authors);
+
+            var resultsWithoutTestAuthor = authorResults
+                .Any(x => !x.ToLower().Contains(Constants.TestPackageAuthor.ToLower()));
+
+            Assert.False(resultsWithoutTestAuthor);
+        }
+
+        [Fact]
+        public async Task ResultsMatchOwnersQueryFieldSearch()
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "owners:" + Constants.TestPackageOwner });
+
+            Assert.NotNull(results);
+
+            var ownerResults = results
+                .Data
+                .Select(x => x.PackageRegistration.Owners);
+
+            var resultsWithoutTestOwner = ownerResults
+                .Any(x => !x.Contains(Constants.TestPackageOwner, StringComparer.OrdinalIgnoreCase));
+
+            Assert.False(resultsWithoutTestOwner);
+        }
+
+        [Fact]
+        public async Task ResultsMatchTagsQueryFieldSearch()
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "tags:" + Constants.TestPackageTag });
+
+            Assert.NotNull(results);
+
+            var tagResults = results
+                .Data
+                .Select(x => x.Tags);
+
+            var resultsWithoutTestTags = tagResults
+                .Any(x => !x.ToLower().Contains(Constants.TestPackageTag.ToLower()));
+
+            Assert.False(resultsWithoutTestTags);
+        }
+
+        public static bool IsPrerelease(string original)
+        {
+            NuGetVersion nugetVersion;
+            if (!NuGetVersion.TryParse(original, out nugetVersion))
```

Comment:
Nit: you can simplify these two lines into

```suggestion
            if (!NuGetVersion.TryParse(original, out var nugetVersion))
```